site/docs/hosting/firebase.md

```diff
@@ -0,0 +1,56 @@
+# Hosting: Firebase Functions
+
+This tutorial will guide you on how to deploy your bot to [firebase](https://firebase.google.com/) by using [Firebase Serverless Functions](https://firebase.google.com/docs/functions).
+
+## Prerequisites
+
+1. Google account (you can [create one here](https://accounts.google.com/signup)).
+
+## Setup
+
+1. [Create a firebase projet](https://firebase.google.com/docs/functions/get-started#create-a-firebase-project)
+2. [Install firebase cli](https://firebase.google.com/docs/functions/get-started#set-up-node.js-and-the-firebase-cli)
+3. [Initialize the project](https://firebase.google.com/docs/functions/get-started#initialize-your-project). You don't have to run step 3. in their docs (`firebase init firestore`). Thats for the their database.
+4. [set up the firebase emulator](https://firebase.google.com/docs/functions/get-started#emulate-execution-of-your-functions) to run your cloud function locally.
+
+## Preparing Your Code
+
+You can use this short example bot as a starting point:
+
+```ts
+import * as functions from "firebase-functions";
+import {webhookCallback, Bot} from "grammy";
+
+const bot = new Bot(<BOT_TOKEN>);
+
+bot.command("start", (ctx) => ctx.reply("Welcome! Up and running."));
+bot.command("ping", (ctx) => ctx.reply(`Pong! ${new Date()}`));
+
+//you can trigger your function on https://localhost/<firebase-projectname>/us-central1/helloworld during development 
```

Comment:
Hmm... I still can't understand it. This doesn't clarify the meaning of the previous statement. Maybe we can create a new paragraph on what it means to trigger a function and why it needs to be done.

Reply:
ah ok, how about "every time you call the url https.//localhost/..... the function will bei executed.
 And additionally  a paragraph explaining how to trigger the function with the telegram bot api  during development (i.e. using ngrok or something to expose the localhost port and setting the webhook)

Reply:
Sounds good!